rgw/v2/tests/s3_swift/test_bucket_lifecycle_object_expiration_transition.py

```diff
@@ -226,8 +228,53 @@ def test_exec(config, ssh_con):
                         upload_start_time,
                         upload_end_time,
                     )
-                    time.sleep(30)
-                    lc_ops.validate_prefix_rule(bucket, config)
+                    if config.test_ops.get("reverse_transition", False):
+                        log.info(f"verifying lc reverse transition")
+                        rule1_lc_seconds = (
+                            config.rgw_lc_debug_interval
+                            * config.test_ops.get("actual_lc_days")
+                        )
+                        rule1_lc_timestamp = upload_end_time + 60 + rule1_lc_seconds
+                        expected_storage_class = config.storage_class
+                        config.test_ops[
+                            "expected_storage_class"
+                        ] = expected_storage_class
+                        lc_ops.validate_prefix_rule(bucket, config)
+
+                        rule2_lc_seconds = (
+                            config.rgw_lc_debug_interval
+                            * config.test_ops.get("rule2_lc_days")
+                        )
+                        rule2_lc_timestamp = rule1_lc_timestamp + rule2_lc_seconds
+                        log.info(
+                            f"sleeping till {datetime.fromtimestamp(rule2_lc_timestamp)} before verifying lc transition rule2"
+                        )
+                        while time.time() < rule2_lc_timestamp:
+                            time.sleep(5)
+                        expected_storage_class = config.second_storage_class
+                        config.test_ops[
+                            "expected_storage_class"
+                        ] = expected_storage_class
+                        lc_ops.validate_prefix_rule(bucket, config)
+
+                        rule3_lc_seconds = (
+                            config.rgw_lc_debug_interval
+                            * config.test_ops.get("rule3_lc_days")
+                        )
+                        rule3_lc_timestamp = rule2_lc_timestamp + rule3_lc_seconds
+                        log.info(
+                            f"sleeping till {datetime.fromtimestamp(rule3_lc_timestamp)} before verifying lc transition rule3"
+                        )
+                        while time.time() < rule3_lc_timestamp:
+                            time.sleep(5)
+                        expected_storage_class = config.storage_class
+                        config.test_ops[
+                            "expected_storage_class"
+                        ] = expected_storage_class
+                        lc_ops.validate_prefix_rule(bucket, config)
+                    else:
+                        time.sleep(30)
```

Comment:
@hmaheswa , what is the reason for adding sleep here?

Reply:
It would be good to add the reason for sleep/wait in the log.info

Reply:
@viduship I have not added this sleep, I just placed what was already there under else block